### PR TITLE
fix(notification) : user comment subscription for notification

### PR DESCRIPTION
**Pull request by yacineKahlerras**
- user that comments is subscribed to the idea card not the comment
- when user comments all who are subscribed to the idea card get a notification

### CODE REVIEW

File: src/api/comment/content-types/comment/lifecycles.js [hunk 1]
```diff
@@ -19,23 +19,32 @@ module.exports = {
...
-      data: {
-        entityType: "Comment",
-        entityId: commentId,
-        createdDateTime: new Date(),
-        active: true,
-        user: user,
-      },
-    });
+    /** verify that user isn't already subscribed to ideaCard */
+    const existingSubscriptions = await strapi.entityService.findMany(
```
Comment (chungthuang) [inline src/api/comment/content-types/comment/lifecycles.js:23]: Do we need to find all of them? Or is `findOne` enough?

File: src/api/comment/content-types/comment/lifecycles.js [hunk 2]
```diff
@@ -19,23 +19,32 @@ module.exports = {
...
+      });
 
     strapi.entityService.create("api::event.event", {
       data: {
         action: "Commented",
         entityName: idea.ideaName,
         content: commentText,
         entityType: "Comment",
-        entityId: commentId,
+        entityId: idea.id,
```
Comment (chungthuang) [inline src/api/comment/content-types/comment/lifecycles.js:47]: Can you add an integration test that would catch this error?

**Comment by chungthuang** [comment]
Can you update the commit message to follow semantic release format https://github.com/semantic-release/semantic-release#commit-message-format? For this PR, it will start with something like `fix(notification)`.

File: src/api/comment/content-types/comment/lifecycles.js (hunk 1 above)
Comment (yacineKahlerras) [reply src/api/comment/content-types/comment/lifecycles.js:23]: yes one is enough, i'll limit it to one entry

File: src/api/comment/content-types/comment/lifecycles.js (hunk 2 above)
Comment (yacineKahlerras) [reply src/api/comment/content-types/comment/lifecycles.js:47]: okay

**Comment by yacineKahlerras** [comment]
@chungthuang updated the code as requested

File: src/api/comment/content-types/comment/lifecycles.js (hunk 2 above)
Comment (chungthuang) [reply src/api/comment/content-types/comment/lifecycles.js:48]: Yes.

**Comment by chungthuang** [comment]
@yacineKahlerras don't forget to fix the commit message and squash the commits.

**Comment by yacineKahlerras** [comment]
@chungthuang done✔

File: tests/event.spec.ts [hunk 3]
```diff
@@ -1,101 +1,111 @@
...
+      );
+
+    // Schema Checks
+    expect(event).toBeDefined(); // Ensure the event exists
+    expect(event.action).toBe("Commented"); // Check the action is "Commented"
+    expect(event.entityName).toBe("This is a test comment"); // The entityName should be the comment text
+    expect(event.content).toBe("This is a test comment"); // Check the event content
+    expect(event.entityType).toBe("Comment"); // Ensure the entityType is "Comment"
+    expect(event.entityId).toBe(ideaId); // The entityId should be the ideaId related to the comment
+    expect(event.eventUser.username).toBe(config.user.username); // Check the eventUser is the comment author
```
Comment (chungthuang) [inline tests/event.spec.ts:109]: Will there also be an event for the idea owner, to notify someone commented on their idea?

**Review by chungthuang — APPROVED** [review]

File: tests/event.spec.ts (hunk 3 above)
Comment (yacineKahlerras) [reply tests/event.spec.ts:109]: @chungthuang no new event is created for the idea owner, when this **comment event** is created, the user and all who are already _subscribed_ to the idea (idea owner, users who commented) get notified

Comment (chungthuang) [reply tests/event.spec.ts:109]: Ok that makes sense. Can this test verify that both the idea owner and commenter get notified?

**Review by chungthuang — APPROVED** [review]

File: tests/event.spec.ts (hunk 3 above)
Comment (yacineKahlerras) [reply tests/event.spec.ts:109]: @chungthuang no because the comment is made by the idea owner and he cant get notified multiple time for the same event

Comment (chungthuang) [reply tests/event.spec.ts:109]: That sounds like we should use a different user to comment to make this test more realistic?

Comment (yacineKahlerras) [reply tests/event.spec.ts:109]: yes we can login with **user2** and comment and then fetch the notifications resulting from that event

**Pull request merged**